pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -2330,6 +2608,76 @@ func TestApplyTaskResults_MinimalExpression(t *testing.T) {
 					}}},
 			},
 		}},
+	}, {
+		name: "Test result substitution on minimal variable substitution expression - matrix include",
+		resolvedResultRefs: ResolvedResultRefs{{
+			Value: *v1beta1.NewStructuredValues("aResultValue"),
+			ResultReference: v1beta1.ResultRef{
+				PipelineTask: "aTask",
+				Result:       "a.Result",
+			},
+			FromTaskRun: "aTaskRun",
+		}},
+		targets: PipelineRunState{{
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:    "bTask",
+				TaskRef: &v1beta1.TaskRef{Name: "bTask"},
+				Matrix: &v1beta1.Matrix{
+					Include: []v1beta1.MatrixInclude{{
+						Name: "build-1",
+						Params: []v1beta1.Param{{
+							Name: "bParam", Value: *v1beta1.NewStructuredValues(`$(tasks.aTask.results["a.Result"])`)}},
+					}}},
+			},
+		}},
+		want: PipelineRunState{{
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:    "bTask",
+				TaskRef: &v1beta1.TaskRef{Name: "bTask"},
+				Matrix: &v1beta1.Matrix{
+					Include: []v1beta1.MatrixInclude{{
+						Name: "build-1",
+						Params: []v1beta1.Param{{
+							Name: "bParam", Value: *v1beta1.NewStructuredValues("aResultValue")}},
+					}}},
+			},
+		}},
+	}, {
+		name: "Test array indexing result substitution on minimal variable substitution expression - matrix include",
```

Comment:
do we support refer an object result value? For include we support string replacements.   string replacements may come from 1. string results 2. array indexing from array result 3. object value from object results. 
Looks like 1 and 2 are tested here.

Reply:
Added test for string replacement from object result